pkg/stanza/fileconsumer/file_sort.go

```diff
@@ -0,0 +1,243 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+)
+
+const (
+	SortTypeNumeric      = "numeric"
+	SortTypeTimestamp    = "timestamp"
+	SortTypeAlphabetical = "alphabetical"
+)
+
+type SortRule interface {
```

Comment:
```suggestion
type sortRule interface {
```
Correct me if I'm wrong, but I think this doesn't need to be exported either.

Reply:
unexported this